src/runtime.rs

```diff
@@ -38,6 +41,8 @@ pub struct Runtime {
 
     /// Shared state that is accessible via all runtimes.
     shared_state: Arc<SharedState>,
+
+    on_cancelation_check: Option<Box<dyn Fn() + RefUnwindSafe + Send>>,
```

Comment:
Hm, why do we need this? I imagined this can work the same way as other events, by user overriding `on_salsa_event` or some such.

Reply:
https://github.com/salsa-rs/salsa/blob/b3cb3c42f258fabf61b930c9600c59dc6ab0ba02/src/lib.rs#L67-L73

Reply:
That needs a `Database`, which `unwind_if_canceled` doesn't have. Should we pass the db as an argument here?

Reply:
Hm, perhaps we should make `unwind_if_canceled` a method of the `Database` then? That would actually give a nice way to opt-out of unwinding -- overrride it to be no-op. Not sure if I love or hate that flexibility though.